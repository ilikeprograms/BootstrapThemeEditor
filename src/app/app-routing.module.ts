import { RouterModule } from '@angular/router';

export const appRoutes = RouterModule.forRoot([{
  path: '',
  pathMatch: 'full',
  redirectTo: 'editor'
}, {
  path: 'editor',
  loadChildren: './page-components/editor/editor.module#EditorModule'
}]);
