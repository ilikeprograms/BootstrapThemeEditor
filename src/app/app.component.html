<clr-main-container>
  <app-header></app-header>

  <div class="content-container">
    <div class="content-area">
      <router-outlet></router-outlet>
    </div>

    <!-- <clr-vertical-nav
      [clrVerticalNavCollapsible]="true"
    >
      <div class="nav-content">
        <clr-vertical-nav-group>
            <clr-icon shape="flame" clrVerticalNavIcon></clr-icon>
            Dr T.
            <clr-vertical-nav-group-children>
              <a clrVerticalNavLink>Charmander</a>
              <a clrVerticalNavLink>Jigglypuff</a>
              <a clrVerticalNavLink>Pikachu</a>
              <a clrVerticalNavLink>Raichu</a>
              <a clrVerticalNavLink>Snorlax</a>
              <div class="nav-divider"></div>
              <a clrVerticalNavLink>Credit</a>
            </clr-vertical-nav-group-children>
        </clr-vertical-nav-group>
      </div>
    </clr-vertical-nav> -->
  </div>

  <!-- <app-social-links></app-social-links> -->
</clr-main-container>
