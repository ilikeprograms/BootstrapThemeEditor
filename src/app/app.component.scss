:host {
  .content-area {
    padding: 0;
  }
}
