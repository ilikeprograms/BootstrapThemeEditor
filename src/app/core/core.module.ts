import { NgModule } from '@angular/core';

import { SassService } from './sass-js/sass.service';

@NgModule({
  providers: [
    SassService
  ]
})
export class CoreModule {}
