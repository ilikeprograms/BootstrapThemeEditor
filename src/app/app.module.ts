import { BrowserModule } from '@angular/platform-browser';
import { NgModule } from '@angular/core';
import { BrowserAnimationsModule } from '@angular/platform-browser/animations';

import { ClarityModule } from '@clr/angular';
import { StoreModule } from '@ngrx/store';
import { StoreDevtoolsModule } from '@ngrx/store-devtools';
import { EffectsModule } from '@ngrx/effects';

import { AppComponent } from './app.component';
import { appRoutes } from './app-routing.module';
import { LayoutModule } from './layout/layout.module';
import { CoreModule } from './core/core.module';
import { indexReducer } from './ngrx/bootstrap/index-reducer';
import { BootstrapFacade } from './ngrx/bootstrap/bootstrap.facade';
import { BootstrapEffects } from './ngrx/bootstrap/bootstrap.effect';

@NgModule({
  declarations: [
    AppComponent
  ],
  imports: [
    BrowserModule,
    ClarityModule,
    BrowserAnimationsModule,
    CoreModule.forRoot(),
    LayoutModule,
    StoreModule.forRoot({}),
    StoreModule.forFeature('bootstrap', indexReducer),
    StoreDevtoolsModule.instrument({
      maxAge: 25, // Retains last 25 states
    }),
    EffectsModule.forRoot([BootstrapEffects]),
    appRoutes
  ],
  providers: [
    BootstrapFacade
  ],
  bootstrap: [AppComponent]
})
export class AppModule { }
