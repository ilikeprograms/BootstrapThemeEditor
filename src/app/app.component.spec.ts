import { TestBed, async } from '@angular/core/testing';
import { RouterTestingModule } from '@angular/router/testing';

import { AppComponent } from './app.component';
import { MockAppHeaderComponent } from 'test/mock/components/app-header.component.mock';
import { MockClrVerticalNavComponent } from 'test/mock/clarity/clr-vertical-nav.component.mock';
import { MockClrIconComponent } from 'test/mock/clarity/clr-icon.component.mock';
import { MockClrVerticalNavGroupChildrenComponent } from 'test/mock/clarity/clr-vertical-nav-group-children.component.mock';
import { MockClrVerticalNavGroupComponent } from 'test/mock/clarity/clr-vertical-nav-group.component.mock';
import { MockClrMainContainerComponent } from 'test/mock/clarity/clr-main-container.component.mock';

describe('AppComponent', () => {
  beforeEach(async(() => {
    TestBed.configureTestingModule({
      imports: [
        RouterTestingModule
      ],
      declarations: [
        AppComponent,
        MockClrMainContainerComponent,
        MockClrIconComponent,
        MockClrVerticalNavComponent,
        MockClrVerticalNavGroupChildrenComponent,
        MockClrVerticalNavGroupComponent,
        MockAppHeaderComponent
      ],
    }).compileComponents();
  }));

  it('should match snapshot', () => {
    const fixture = TestBed.createComponent(AppComponent);
    const app = fixture.debugElement.componentInstance;

    expect(fixture.debugElement.nativeElement).toMatchSnapshot();
  });
});
