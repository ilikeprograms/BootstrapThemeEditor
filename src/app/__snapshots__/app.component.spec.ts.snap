// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppComponent should match snapshot 1`] = `
<div
  id="root0"
  ng-version="7.2.9"
>
  <clr-main-container>
    <app-header />
    <div
      class="content-container"
    >
      <div
        class="content-area"
      >
        <router-outlet />
      </div>
      <clr-vertical-nav>
        <clr-vertical-nav-group>
          <clr-icon
            clrverticalnavicon=""
            shape="flame"
          />
           Dr T. 
          <clr-vertical-nav-group-children>
            <a
              clrverticalnavlink=""
            >
              Charmander
            </a>
            <a
              clrverticalnavlink=""
            >
              Jigglypuff
            </a>
            <a
              clrverticalnavlink=""
            >
              Pikachu
            </a>
            <a
              clrverticalnavlink=""
            >
              Raichu
            </a>
            <a
              clrverticalnavlink=""
            >
              Snorlax
            </a>
            <div
              class="nav-divider"
            />
            <a
              clrverticalnavlink=""
            >
              Credit
            </a>
          </clr-vertical-nav-group-children>
        </clr-vertical-nav-group>
      </clr-vertical-nav>
    </div>
  </clr-main-container>
</div>
`;
